src/app/components/calculator/calculator.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import { Calculator } from 'src/app/classes/calculator';
+import { History } from 'src/app/classes/history';
+import { CoursesService } from 'src/app/services/courses.service';
+
+@Component({
+  selector: 'app-calculator',
+  templateUrl: './calculator.component.html',
+  styleUrls: ['./calculator.component.scss']
+})
+export class CalculatorComponent implements OnInit {
+  totalHours: number | undefined;
+  hoursPerWeek: number | undefined;
+
+  constructor(private coursesService: CoursesService) { }
+
+  ngOnInit(): void {
+  }
+
+  onSubmit(form: History){
+    form.selectedCourses = (this.coursesService.selectedCourses.map(c => c.name)).toString();
+    form.totalHours =  this.coursesService.selectedCourses.map(c => c.hours).reduce((total, hours) => total! + hours!, 0);
+    let weeks = this.getWeeksDiff(form)
+    form.hoursPerWeek = form.totalHours!/weeks;
+    this.hoursPerWeek = form.hoursPerWeek;
+    this.totalHours = form.totalHours;
+    this.coursesService.addHistory(form).subscribe((data)=>{
```

Comment:
toda vez que vc usa .subscribe() vc tem que fazer um .unsubscribe() depois, senao o angular vai guardar isso na memoria pra sempre e se vc nao vai fazer nada com o retorno entao nao precisa fazer o arrow function

```suggestion
    const addHistorySub = this.coursesService.addHistory(form).subscribe();
    this.openSubscriptions.push(addHistorySub )
```